src/Common/ZooKeeper/ZooKeeper.cpp

```diff
@@ -1057,7 +1087,7 @@ std::future<Coordination::ListResponse> ZooKeeper::asyncTryGetChildrenNoThrow(
         promise->set_value(response);
     };
 
-    impl->list(path, list_request_type, std::move(callback), watch_callback);
+    impl->list(path, list_request_type, std::move(callback), std::move(watch_callback));
```

Comment:
Here you explicitly moved the `Coordination::WatchCallbackPtr`, but not in other places. Is there any particular reason for this difference? I don't think it should matter whether we copy or move, but I think we should be consistent.